teuthology/task/install/__init__.py

```diff
@@ -565,11 +565,15 @@ def task(ctx, config):
     log.debug('project %s' % project)
     overrides = ctx.config.get('overrides')
     repos = None
+    extra_system_packages = config.get('extra_system_packages', [])
+
     if overrides:
         install_overrides = overrides.get('install', {})
+        log.debug('INSTALL overrides: %s' % install_overrides)
         teuthology.deep_merge(config, install_overrides.get(project, {}))
+        teuthology.deep_merge(extra_system_packages, install_overrides.get('extra_system_packages', {}))
```

Comment:
It looks like this completely wrecked the RBD suite:
```
2024-05-17T01:31:22.228 INFO:teuthology.run_tasks:Running task install...
2024-05-17T01:31:22.236 DEBUG:teuthology.task.install:project ceph
2024-05-17T01:31:22.236 DEBUG:teuthology.task.install:INSTALL overrides: {'ceph': {'flavor': 'default', 'sha1': 'c2cedb25d8b4c1afc8e737ae85036a985a644b58'}}
2024-05-17T01:31:22.236 ERROR:teuthology.run_tasks:Saw exception from tasks.
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_teuthology_a771c3cd04bbdbdd87e898bc8a2abe9d693c14db/teuthology/run_tasks.py", line 109, in run_tasks
    manager.__enter__()
  File "/usr/lib/python3.10/contextlib.py", line 135, in __enter__
    return next(self.gen)
  File "/home/teuthworker/src/git.ceph.com_teuthology_a771c3cd04bbdbdd87e898bc8a2abe9d693c14db/teuthology/task/install/__init__.py", line 574, in task
    teuthology.deep_merge(extra_system_packages, install_overrides.get('extra_system_packages', {}))
  File "/home/teuthworker/src/git.ceph.com_teuthology_a771c3cd04bbdbdd87e898bc8a2abe9d693c14db/teuthology/misc.py", line 1012, in deep_merge
    assert isinstance(b, list)
AssertionError
2024-05-17T01:31:22.411 ERROR:teuthology.util.sentry: Sentry event: https://sentry.ceph.com/organizations/ceph/?query=50429c2dc2cd462395434a83df45a4db
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_teuthology_a771c3cd04bbdbdd87e898bc8a2abe9d693c14db/teuthology/run_tasks.py", line 109, in run_tasks
    manager.__enter__()
  File "/usr/lib/python3.10/contextlib.py", line 135, in __enter__
    return next(self.gen)
  File "/home/teuthworker/src/git.ceph.com_teuthology_a771c3cd04bbdbdd87e898bc8a2abe9d693c14db/teuthology/task/install/__init__.py", line 574, in task
    teuthology.deep_merge(extra_system_packages, install_overrides.get('extra_system_packages', {}))
  File "/home/teuthworker/src/git.ceph.com_teuthology_a771c3cd04bbdbdd87e898bc8a2abe9d693c14db/teuthology/misc.py", line 1012, in deep_merge
    assert isinstance(b, list)
AssertionError
```
https://pulpito.ceph.com/yuriw-2024-05-17_01:07:43-rbd-wip-yuri6-testing-2024-05-16-0833-reef-distro-default-smithi/

Reply:
I have created a ticket for this: https://tracker.ceph.com/issues/66093
I think the following change should work, I can test and open PR:
```suggestion
        teuthology.deep_merge(extra_system_packages, install_overrides.get(project, {}))
```

Reply:
opened PR: https://github.com/ceph/teuthology/pull/1942